libnavigation-core/src/test/java/com/mapbox/navigation/core/fasterroute/RejectedRoutesTrackerTest.kt

```diff
@@ -0,0 +1,67 @@
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.directions.session.RoutesExtra
+import com.mapbox.navigation.core.directions.session.RoutesUpdatedResult
+import com.mapbox.navigation.core.fasterroute.TestRoutes.Companion.MUNICH_NUREMBERG
+import com.mapbox.navigation.testing.LoggingFrontendTestRule
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+class RejectedRoutesTrackerTest {
+
+    @get:Rule
+    val loggerRule = LoggingFrontendTestRule()
+
+    @Test
+    fun `track routes from Munich to Nuremberg moving by the slowest route`() = runBlocking<Unit> {
+        val rejectedRoutesTracker = createRejectedRoutesTracker()
+        val recordedRoutesUpdates = readRouteObserverResults(MUNICH_NUREMBERG)
+        val untrackedRoutesIds = mutableListOf<String>()
+        recordedRoutesUpdates.forEachIndexed { index, recordedRoutesUpdateResult ->
+            val routesUpdate = recordedRoutesUpdateResult.update
+            if (routesUpdate.reason == RoutesExtra.ROUTES_UPDATE_REASON_NEW) {
+                val alternatives = createAlternativesMap(routesUpdate, recordedRoutesUpdateResult)
+                rejectedRoutesTracker.clean()
+                rejectedRoutesTracker.addRejectedRoutes(alternatives)
+            }
+            if (routesUpdate.reason == RoutesExtra.ROUTES_UPDATE_REASON_ALTERNATIVE) {
+                val alternatives = createAlternativesMap(routesUpdate, recordedRoutesUpdateResult)
+                val result = rejectedRoutesTracker.findUntrackedAlternatives(alternatives)
+                untrackedRoutesIds.addAll(result.map { it.id })
+            }
+        }
+        assertEquals(
+            listOf(
+                // this one feels like similar to initial alternative because they both go through A9
+                "UOIW_1UUIDFfyICssWNnKB2o4cANnHc5pQ4WjsBOKW694GD7ZFwG5Q==#1",
```

Comment:
I'm thing about comparing routes not only by geometry, but also using summary see `calculateDescriptionSimilarity`. That would help with the cases when routes have not super similar geometry, but follow the same highway for example. The main thing that stop me is the fact that `summary` filed from route leg doesn't have stable format.

Reply:
@VysotskiVadim have you considered using `LegStep#ref` or `LegStep#name`? This paired with the `distance` annotation (or any other mean to create a ratio of distance traveled per `ref`/`name`) should give us an understanding of the road network that is used for a specific route.